llvm/test/tools/llvm-objdump/ELF/dynamic-malformed.test

```diff
@@ -29,10 +28,36 @@ FileHeader:
   Class:   ELFCLASS64
   Data:    ELFDATA2LSB
   Type:    ET_EXEC
-  Machine: EM_X86_64
 Sections:
   - Name: .dynamic
     Type: SHT_DYNAMIC
     Entries:
       - Tag:   DT_SONAME
         Value: 1
+
+# RUN: yaml2obj %s --docnum=3 -o %t.invalidaddr
+# RUN: llvm-objdump -p %t.invalidaddr | FileCheck %s --check-prefix=ADDR
+# RUN: llvm-objdump -p %t.invalidaddr 2>&1 >/dev/null | \
+# RUN: FileCheck %s -DFILE=%t.invalidaddr --implicit-check-not=warning --check-prefix=ADDR-ERROR
```

Comment:
Assuming what you're trying to achieve here is check stdout and stderr independently, I wouldn't bother: we usually just check them all in the combined output, so I'd write things like this:

```
# RUN: llvm-objdump -p %t.invalidaddr 2>&1 | \
# RUN:   FileCheck %s --check-prefix=ADDR --implicit-check-not-warning=warning:
```
and then just reorder the check lines accordingly. A couple of other minor notes from that by the way: a) I suggest including the colon after the implicit-check-not pattern, to reduce the risk of spurious matches against things like file paths; b) I find it more readable to break lines in the manner illustrated (`| \` at the end of the first line and additional spaces at the start of the second).

Reply:
Okay, the printing order is now fixed, but upon further investigation, it looks like the original bug is not fixed now.